src/host.ts

```diff
@@ -41,13 +125,13 @@ export interface HostBaseOptions {
 export abstract class HostBase {
     public readonly ignoreCase: boolean;
 
-    private useBuiltinGrammars: boolean;
-    private innerParser: Parser | undefined;
-    private knownGrammars: Map<string, string> | undefined;
+    private _innerParser: Parser | undefined;
+    private _knownGrammars: Map<string, string> | undefined;
+    private _useBuiltinGrammars: boolean;
 
-    constructor({ ignoreCase = ignoreCaseFallback, knownGrammars, useBuiltinGrammars = true }: HostBaseOptions = {}) {
+    constructor({ ignoreCase = false, knownGrammars, useBuiltinGrammars = true }: HostBaseOptions = {}) {

```

Comment:
Built-in grammars don't make all that much sense for non-Node hosts, at least as currently implemented. Does it need to be part of the base API?

Reply:
Built-in grammars are part of the package and have no dependency on Node being the host. That said, the built-in grammars are fairly old at this point. I'm using a slightly different mechanism in ecmarkup-vscode for writing derived grammars (essentially fetching the grammar from the ecma262 repo). I haven't decided what to do with this yet and will probably leave it in until I plan out what 3.x should look like.

Reply:
> Built-in grammars are part of the package and have no dependency on Node being the host.

Well, no, but their implementation kind of does: passing `useBuiltinGrammars = true` will [make](https://github.com/rbuckton/grammarkdown/blob/5a86b9fb55d00e4ed2a87d37d55b7723dc428b67/src/host.ts#L560) `resolveFile('ES6')` return the path on disk to `es2015.grammar`, but that path is only meaningful on Node hosts.

But if you're planning on revisiting it later anyway it makes sense to ignore it for now.

Reply:
It will return the result of `require.resolve`, which (IIRC) bundlers like WebPack overwrite.

Reply:
Sure, it's just that a host which is not going to do some equivalent of `fs.readFile` can't really do much useful with those paths.